Unity/Champloo/Assets/Scripts/Player/Movement States/Movement Specials/OnInfiniteDash.cs

```diff
@@ -36,8 +44,9 @@ public override MovementState DecideNextState(Vector3 velocity, Vector3 external
         if (isDisabled || timingState == TimingState.DONE)
         {
             hitDashEarly = false;
-            if (earlyAttackInput)
+            if (attackAllowed)
```

Comment:
That would be better; but I'm not sure we event need this logic anymore. From what I've seen, the Player's attack logic doesn't short circuit on state anymore.